stdlib/Dates/src/periods.jl

```diff
@@ -325,7 +325,7 @@ end
 Base.show(io::IO,x::CompoundPeriod) = print(io, string(x))
 
 Base.convert(::Type{T}, x::CompoundPeriod) where T<:Period =
-    isconcretetype(T) ? sum(T, x.periods) : throw(MethodError(convert,(T,x)))
+    isconcretetype(T) ? sum(T, x.periods; init = zero(T)) : throw(MethodError(convert,(T,x)))
```

Comment:
It seems like `sum(T::Type, iterator)` should use `zero(T)` automatically for empty collections.   Wouldn't that be a better fix?

Reply:
In particular, we could define
```jl
Base.mapreduce_empty(::Type{T}, ::typeof(Base.add_sum), ::Any) where {T} = zero(T)
Base.reduce_empty_iter(::Base.MappingRF{Type{T},typeof(Base.add_sum)}, ::Base.EltypeUnknown) where {T} = zero(T)
```

Reply:
ref #36262

Reply:
Oh, this option was discussed and rejected in #20561.